### PR TITLE
fix(overlay): incorrect bounding box styles if position is exactly zero

### PR DESCRIPTION
**Pull request by crisbeto**
Handles the edge case where the bounding box styles won't be set correctly, causing the overlay to jump to a wrong position, if either of its values works out to exactly zero.

### CODE REVIEW

File: src/cdk/overlay/position/flexible-connected-position-strategy.spec.ts [hunk 1]
```diff
@@ -1119,6 +1119,102 @@ describe('FlexibleConnectedPositionStrategy', () => {
...
+          originX: 'start'
+        }]);
+
+      attachOverlay({positionStrategy});
+
+      const boundingBox = overlayContainer
+        .getContainerElement()
+        .querySelector('.cdk-overlay-connected-position-bounding-box') as HTMLElement;
+
+      expect(boundingBox.style.bottom).toBe('0px');
```
Comment (jelbourn) [inline src/cdk/overlay/position/flexible-connected-position-strategy.spec.ts]: Add comment for these tests like "Ensure that `0px` is explicitly set rather than the property being left blank due to zero being falsy."
